### ISSUE TITLE
Software Elemets

### ISSUE THREAD

**Issue by kborovik** [logictimes/PrivateKey.Space#16]
**Bitcoin Network**
Bitcoin Core

**Back-End**
Lib:

**Front-End**
Framework:

**Comment by alienatix** [comment]
1. We'd recommend using PostgreSQL instead of MongoDB. Main reasons are:
* it supports JSONB format
* it provides better performance
* as [this article](https://www.sisense.com/blog/postgres-vs-mongodb-for-storing-json-data/) mentions 
> MongoDB is ideal for unstructured data, but if you plan to move to a mix of structured and unstructured data later on, or if you think ACID compliance might become important in the future, Postgres may work best.

2. Not sure that we need Libbitcoin
3. Trezor Link is a low-level library. From github:
> You probably don't want to use this package directly. For communication with Trezor with a more high-level API, use trezor.js

The difference between Trezor Connect and trezor.js is that the second one allows UI cutomization. That's why I'd stick to trezor.js

4. To Front-End section please add redux lib and redux-observable middleware

**Comment by kborovik** [comment]
@alienatix, done.

**Comment by alienatix** [comment]
While discussing back-end framework internally we came to the conclusion that [Koa](http://koajs.com/) would fit this project best. If you don't have any objection, please add it to the requirements.

**Comment by kborovik** [comment]
@alienatix just to be clear, you are suggesting to replace React with Koa, right?

**Comment by alienatix** [comment]
Not really, Koa is a Node.js back-end framework (an alternative to [Express](https://expressjs.com/))
